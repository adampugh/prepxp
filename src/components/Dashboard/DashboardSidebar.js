import React from "react";
import { Link } from "react-router-dom";

import user from "../../assets/images/default-user-image.png";

const dashboardSidebar = (props) => (
    <div className="sidebar">
        <div className="sidebar__img">
            <img src={user} alt="default user" />
            <h1>{props.name}</h1>
            <hr />
        </div>
        <div className="sidebar__grid">
            <h2><i className="fas fa-tachometer-alt"></i></h2>
            <Link to="/prepxp/dashboard"><h2 className="sidebar__grid__link">Dashboard</h2></Link>
            <h2><i className="fas fa-pencil-alt"></i></h2>
            <h2 onClick={props.openCreateModal} className="sidebar__grid__link">Create</h2>
            <h2><i className="fas fa-search"></i></h2>
            <Link to="/prepxp/search"><h2 className="sidebar__grid__link">Search</h2></Link>
            <h2><i className="fas fa-book"></i></h2>
            <Link to="/prepxp/blog"><h2 className="sidebar__grid__link">Blog</h2></Link>
            <h2><i className="fas fa-sign-out-alt"></i></h2>
            <h2 onClick={props.startLogout} className="sidebar__grid__link">Logout</h2>
        </div>
        
    </div>
);

export default dashboardSidebar;