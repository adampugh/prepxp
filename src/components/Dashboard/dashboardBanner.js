import React from "react";

 
const dashboardBanner = (props) => (
    <section className="dashboard__banner">
        <div className="container">
            <p>Create a new list of interview questions</p>
            <button className="btn" onClick={props.openCreateModal}>Create</button>
            
        </div>
    </section>
);

export default dashboardBanner;