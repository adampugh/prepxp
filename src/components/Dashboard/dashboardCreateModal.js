import React, { Component } from "react";
import Modal from "react-modal";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      }
}


class DashboardCreateModal extends Component {
    state = {
        listTitle: "",
        error: false
    }

    handleKeyDown = (e) => {
        if (e.key === "Enter") {
            this.handleStartAddList();
        }
    }

    updateTitle = (e) => {
        let hasError = e.target.value.length < 0
        this.setState({
            listTitle: e.target.value,
            error: hasError
        });
    }

    handleStartAddList = () => {
        let list = {
            title: this.state.listTitle,
            questions: [],
            // remove this once firebase generates ids
            id: Math.random().toString()
        }
        this.state.listTitle.length > 0 
            ? this.props.startAddList(list)
            : this.setState({
                error: true
            })
    }

    
    render() {
        return (
            <Modal
                isOpen={this.props.isOpen}
                appElement={document.getElementById('root')} 
                onRequestClose={this.props.closeModal}
                style={customStyles}
                closeTimeoutMS={200}
                className="modal"
            >
                <div className="modal--delete">
                    <span onClick={this.props.closeModal}><i className="fas fa-times"></i></span>
                    <h2>Enter list title</h2>
                    <hr />
                    {this.state.error 
                        ? <p className="modal__error">Please enter a list title</p>
                        : null
                    }
                    <input 
                        type="text" 
                        onChange={(e) => this.updateTitle(e)} 
                        onKeyDown={(e) => this.handleKeyDown(e)} />
                    <button className="btn" onClick={this.handleStartAddList}>Start</button>
                    <button className="btn" onClick={this.props.closeModal}>Cancel</button>
                </div>
            </Modal>
        )
    }
}

export default DashboardCreateModal;