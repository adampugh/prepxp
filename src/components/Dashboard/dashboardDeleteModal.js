import React, { Component } from "react";
import Modal from "react-modal";

class DashboardDeleteModal extends Component {
    render() {
        return (
            <Modal></Modal>
        )
    }
}

export default DashboardDeleteModal;