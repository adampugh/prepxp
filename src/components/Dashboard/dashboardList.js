import React from "react";
import { Link } from "react-router-dom";

const dashboardList = (props) => (
    <div className="dashboard__grid__lists__item">
        <div>
            <Link to={{pathname:"/list/start", state: {list: props.list}}}>
                <h2>{props.list.title}</h2>
            </Link>
            <p>{props.list.questions.length} Questions</p>
        </div>
        <div>
            <Link to={{pathname:"/list/edit", state: {list: props.list, listIndex: props.listIndex}}}>
                <h2><i className="fas fa-edit"></i></h2>
            </Link>
            <h2 onClick={() => {
                props.selectList(props.id);
                props.openModal();
            }}><i className="fas fa-trash-alt"></i></h2>
        </div>
    </div>
);

export default dashboardList;