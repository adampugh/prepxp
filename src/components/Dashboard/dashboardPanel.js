import React from "react";
import CircularProgressbar from 'react-circular-progressbar';

const dashboardPanel = (props) => {
    let listsCompleted = props.lists.filter(list => list.timesCompleted > 0).length;
    let totalAnsweredPercentage = props.totalQuestions > 0 
        ? Math.round((props.totalAnsweredQuestions / props.totalQuestions) * 100)
        : 0;
    let listsCompletedPercentage = props.lists.length > 0 
        ? Math.round((listsCompleted / props.lists.length) * 100)
        : 0;

    return (
    <div className="dashboard__grid__panel">
        <div className="dashboard__grid__panel__box">
            <div>
                <h2>Questions Answered</h2>
                <CircularProgressbar 
                    percentage={ totalAnsweredPercentage } 
                    initialAnimation={true} 
                    backgroundPadding={10}/>
                <p>{`${props.totalAnsweredQuestions} / ${props.totalQuestions}`}</p>
            </div>
            <div>
                <h2>Lists Completed</h2>
                <CircularProgressbar 
                    percentage={ listsCompletedPercentage } 
                    initialAnimation={true} 
                    backgroundPadding={10} />
                <p>{`${listsCompleted} / ${props.lists.length}`}</p>
            </div>
            <div className="dashboard__grid__panel__box__mostCompleted">
                <h2>Most Completed</h2>
                {
                    props.lists.sort((a, b) => b.timesCompleted - a.timesCompleted)
                        .slice(0, 3)
                        .map((list, index) => 
                            <p key={index + list.title}>{`${index + 1}. ${list.title} `}
                                <span>{`${list.timesCompleted}`}</span>
                            </p>)
                }
            </div>
        </div>
    
    </div>
)
}
export default dashboardPanel;


// props.lists.sort((a, b) => b.timesCompleted - a.timesCompleted)
// .map((list, index) => { if (index < 3) {
//     return <p key={index + list.title}>{`${index + 1}. ${list.title} - ${list.timesCompleted}`}</p>
// }})