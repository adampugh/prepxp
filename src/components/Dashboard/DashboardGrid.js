import React, { Component } from "react";

import DashboardSidebar from "./DashboardSidebar";
import DashboardList from "./dashboardList";

class DashboardGrid extends Component {
    render(props) {
        return (
            <section className="container">
                <div className="dashboard__grid">
                    <DashboardSidebar />
                    <div className="dashboard__grid__lists">
                        <h1>Hello, {this.props.name}!</h1>
                        {this.props.lists.length > 0 ? this.props.lists.map((list, index) => <DashboardList key={`${list.title + index}`} list={list} />)
                            : 
                            <div className="dashboard__empty">
                                <h2>You don't have any lists yet, add or create some!</h2>
                                <button className="btn">Lists</button>
                            </div> 
                        }
                    </div>
                </div>
            </section>
        );
    }
}

export default DashboardGrid;