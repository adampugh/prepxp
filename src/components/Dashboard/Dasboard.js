import React, { Component } from "react";
import Modal from "react-modal";

import Navbar from "../UI/navbarLoggedIn";
import DashboardGrid from "./DashboardGrid";

import Footer from "../UI/footer";

class Dashboard extends Component {
    state = {
        name: "Bob",
        lists: [
            {
                title: "JavaScript Developer",
                questions: [
                    {
                        question: "What is the difference between let and var?",
                        answer: "JavaScript is a language for the web."
                    },
                    {
                        question: "Who created JavaScript?",
                        answer: "Brendan Eich"
                    }
                ]
            },
            {
                title: "PHP Developer",
                questions: [
                    {
                        question: "What is PHP exactly?",
                        answer: "No-one really knows."
                    }
                ]
            }
        ],
        modalIsOpen: false
    }

    componentDidMount() {
        window.scrollTo(0, 0);
    }

    closeModal = () => {
        this.setState({
            modalIsOpen: false
        })
    }

    openModal = () => {
        this.setState({
            modalIsOpen: true
        })
    }

    render() {
        return (
            <div>
                <Modal 
                    isOpen={this.state.modalIsOpen}
                    onRequestClose={this.closeModal}
                />
                <Navbar />
                <DashboardGrid 
                    name={this.state.name} 
                    lists={this.state.lists}
                    openModal={this.openModal}
                    closeModal={this.closeModal} />
                <Footer />
            </div>
        );
    }
}

export default Dashboard;