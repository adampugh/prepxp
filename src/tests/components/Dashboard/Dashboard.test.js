import React from "react";
import { shallow } from "enzyme";

import Dashboard from "../../../components/Dashboard/Dasboard";

describe("<Dashboard />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<Dashboard />);
        expect(wrapper).toMatchSnapshot();
    });

    // check delete function returns action obj
});