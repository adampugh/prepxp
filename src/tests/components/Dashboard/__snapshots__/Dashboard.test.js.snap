// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should match snapshot 1`] = `
<div>
  <DashboardDeleteModal
    closeModal={[Function]}
    isOpen={false}
    selectedListId=""
  />
  <DashboardCreateModal
    closeModal={[Function]}
    isOpen={false}
  />
  <Navbar />
  <dashboardBanner
    openCreateModal={[Function]}
  />
  <DashboardGrid
    closeCreateModal={[Function]}
    closeDeleteModal={[Function]}
    lists={
      Array [
        Object {
          "id": "234",
          "questions": Array [
            Object {
              "answer": "JavaScript is a language for the web.",
              "question": "What is the difference between let and var?",
            },
            Object {
              "answer": "Brendan Eich",
              "question": "Who created JavaScript?",
            },
          ],
          "title": "JavaScript Developer",
        },
        Object {
          "id": "1231",
          "questions": Array [
            Object {
              "answer": "No-one really knows.",
              "question": "What is PHP exactly?",
            },
          ],
          "title": "PHP Developer",
        },
      ]
    }
    name="Bob"
    openCreateModal={[Function]}
    openDeleteModal={[Function]}
    selectList={[Function]}
  />
  <p>
    Logout
  </p>
  <footer />
</div>
`;
